### DISCUSSION TITLE
[Question]: How to delete user system

### DISCUSSION THREAD

**Discussion by kuangxiaoye** [danny-avila/LibreChat#872] [Q&A] [answered]
### Contact Details

w258765@gmail.com

### What is your question?

The user system is very bloated, and I don't understand why the original user system has now become a must.

### More Details

# ENABLE_USER_SYSTEM=1  # <-- make sure you don't comment this back in if you're not using your own user system

How can I shut down the user system?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
ENABLE_USER_SYSTEM no longer works... please check the new .env.example. For now, you are required to create a user, and there is no way to disable login.

**Comment by danny-avila** [comment, marked as answer]
The project is very tightly coupled to the user system now. I think this elevates its use case for a lot of people, rather than being a standalone one-person-only UI. You have the option of disabling registration to keep it to one or select users only.

**Reply by Enduriel** [reply]
Bit of a necro but I'd like to say that for homelab use, every service I is going to be put behind something like authentik or authelia so being forced to have an extra login interface is frustrating.

**Reply by kuangxiaoye** [reply]
> Bit of a necro but I'd like to say that for homelab use every service I use it going to be put behind something like authentik or authelia so being forced to have an extra login interface is frustrating.

You can see that OpenAI has also deliberately started to avoid logging in hhh

**Reply by danny-avila** [reply]
I can think about how to add a single user mode.

> You can see that OpenAI has also deliberately started to avoid logging in hhh

This is a business strategy on their part to reduce friction for new users. They are essentially paying for training data by not requiring login for chatting.

In any case, it might make sense to offer a "no login" mode like OpenAI, as it has been suggested a few times, where chats are not saved.

This would be separate from a "single user" mode, where chats are saved and no login is provided, with the latter being more of a priority and less involved.

**Reply by CthulhuDen** [reply]
Some more necro posting, but I think it would make more sense to implement "external auth" (authenticating reverse proxy) support, in which librechat reads user properties from headers, auto creating user as needed. This way users and access to apps (including librechat) are in the Authentik/Keycloak/etc, no extra login interface in LibreChat, and still chats are separated by users and are persistent.
